pandas/io/common.py

```diff
@@ -450,12 +452,20 @@ def file_path_to_url(path: str) -> str:
     return urljoin("file:", pathname2url(path))
 
 
+_extension_to_compression = {
+    ".tar": "tar",
+    ".tar.gz": "tar",
+    ".tar.bz2": "tar",
+    ".tar.xz": "tar",
+    ".gz": "gzip",
+    ".bz2": "bz2",
+    ".zip": "zip",
+    ".xz": "xz",
+    ".zst": "zstd",
+}
+_supported_compressions = set(_extension_to_compression.values())
 _compression_to_extension = {
-    "gzip": ".gz",
-    "bz2": ".bz2",
-    "zip": ".zip",
-    "xz": ".xz",
-    "zstd": ".zst",
+    value: key for key, value in _extension_to_compression.items()
```

Comment:
This inverts `_extension_to_compression`? Will this not end up with `".tar": "tar.xz"` (overwriting the previous tar keys?)

Reply:
that's true. `_extension_to_compression` is only used in tests as fixtures, though: https://cs.github.com/?scopeName=All+repos&scope=&q=repo%3Apandas-dev%2Fpandas+_compression_to_extension
since we don't need to test every single variant of `.tar` (that's dealed with by python's builtin `tarfile`), I reckon it's fine to only test for one of the variants.

Reply:
Would be nice to move it then to the tests (if it makes sense). Testing only one of them seems fine with me. Might need a test for the new mode arguments.

Reply:
will do. with the `mode` argument, I'll see if there's a way to *not* add it - there's plenty of ways how users can misunderstand the argument, and adding it for the sake of different tar compression options doesn't seem like good enough of a reason

Reply:
Okay! When seeing `mode`, I mainly think about text vs binary. Some IO functions currently don't have it (technically all of the csv/json should have it - in case the text/binary auto-detection fails).

Reply:
moved `_compression_to_extension` into tests in https://github.com/pandas-dev/pandas/pull/44787/commits/861faf0e3cf86374946b56767f43067da0fde349.